ngraph/core/reference/include/ngraph/runtime/reference/ctc_greedy_decoder.hpp

```diff
@@ -59,8 +59,7 @@ namespace ngraph
                         auto data_index = data_transform.index({seq_ind, batch_ind, 0});
                         auto mask_index = seq_masks_transform.index({seq_ind, batch_ind});
 
-                        // first 0 marks the end of a sequence
-                        if (seq_ind && sequence_masks[mask_index] == T{0})
+                        if (sequence_masks[mask_index] == T{0})
```

Comment:
Could you elaborate on this change? Previously the zero length sequence was not allowed (seq_ind was checked to be non-zero). But now we allow this behaviour. Do you know or guess why the original behavior was used?

Reply:
@iimironov , have what is the behaviour of TF or other frameworks in this case?

Reply:
I have no idea, why here is such restriction. That is not described in specification. I faced with that when I was extending layers tests, due to plugin allows that.

Reply:
@iimironov, please provide feedback.

Reply:
In tf ref not checks that case.  That case not covered specification. I suggest to check how it implemented in kaldi fw and correct ctc_greedy_decoder_v1.

Reply:
Kaldi does not have such layer. @lazarevevgeny , do you have other concerns about that?